CHANGELOG.md

```diff
@@ -2,7 +2,7 @@
 
 Canonical reference for changes, improvements, and bugfixes for Boundary.
 
-## Next
+## 0.16.0 (2024/04/15)
```

Comment:
I haven't actually done much with changelogs in the past, but looking at previous examples, it looks like we leave the `## Next` section in there?

https://github.com/hashicorp/boundary/commit/fa38d37ba791ef2c251e211d90dec05483bbe978